src/contracts/core/AVSDirectory.sol

```diff
@@ -249,114 +265,192 @@ contract AVSDirectory is
     }
 
     /**
-     *  @notice Called by an AVS to emit an `AVSMetadataURIUpdated` event indicating the information has updated.
-     *
-     *  @param metadataURI The URI for metadata associated with an AVS.
-     *
-     *  @dev Note that the `metadataURI` is *never stored* and is only emitted in the `AVSMetadataURIUpdated` event.
+     * @notice Modifies the propotions of slashable stake allocated to a list of operatorSets for a set of strategies
+     * @param operator address to modify allocations for
+     * @param allocations array of magnitude adjustments for multiple strategies and corresponding operator sets
+     * @param operatorSignature signature of the operator if msg.sender is not the operator
+     * @dev updates freeMagnitude for the updated strategies
+     * @dev must be called by the operator
      */
-    function updateAVSMetadataURI(string calldata metadataURI) external override {
-        emit AVSMetadataURIUpdated(msg.sender, metadataURI);
-    }
+    function modifyAllocations(
+        address operator,
+        MagnitudeAllocation[] calldata allocations,
+        SignatureWithSaltAndExpiry calldata operatorSignature
+    ) external {
+        if (msg.sender != operator) {
+            _verifyOperatorSignature(operator, allocations, operatorSignature);
+        }
 
-    /**
-     * @notice Called by an operator to cancel a salt that has been used to register with an AVS.
-     *
-     * @param salt A unique and single use value associated with the approver signature.
-     */
-    function cancelSalt(bytes32 salt) external override {
-        // Mutate `operatorSaltIsSpent` to `true` to prevent future spending.
-        operatorSaltIsSpent[msg.sender][salt] = true;
+        // completable timestamp for deallocations
+        uint32 completableTimestamp = uint32(block.timestamp) + DEALLOCATION_DELAY;
+        // effect timestamp for allocations to take effect. This is configurable by operators
+        uint32 effectTimestamp = uint32(block.timestamp) + getAllocationDelay(operator);
+        for (uint256 i = 0; i < allocations.length; ++i) {
+            // 1. For the given (operator,strategy) clear all pending free magnitude for the strategy and update freeMagnitude
+            // numToComplete = 0 defaults to completing all pending deallocations up to uint8 max (256)
+            _updateFreeMagnitude({operator: operator, strategy: allocations[i].strategy, numToComplete: 0});
+
+            // 2. check current totalMagnitude matches expected value
```

Comment:
Solves race condition with slashing

Reply:
Maybe add comment clarifying this?

Reply:
documenting this is good. otherwise this is an opportunity for someone to identify this reversion as a race condition "issue" rather than a purposeful resolution of another issue.

Reply:
added clarifying note